local-cluster/tests/local_cluster.rs

```diff
@@ -1536,20 +1538,32 @@ fn test_wait_for_max_stake() {
     solana_logger::setup_with_default(RUST_LOG_FILTER);
     let validator_config = ValidatorConfig::default_for_test();
     let slots_per_epoch = MINIMUM_SLOTS_PER_EPOCH;
+    // Set this large enough to allow for skipped slots but still be able to
+    // make a root and derive the new leader schedule in time.
+    let stakers_slot_offset = slots_per_epoch.saturating_mul(MAX_LEADER_SCHEDULE_EPOCH_OFFSET);
+    // Reduce this so that we can complete the test faster by advancing through
+    // slots/epochs faster. But don't make it too small because it can make us
+    // susceptible to skipped slots and the cluster getting stuck.
+    let ticks_per_slot = 16;
     let mut config = ClusterConfig {
         cluster_lamports: DEFAULT_CLUSTER_LAMPORTS,
         node_stakes: vec![DEFAULT_NODE_STAKE; 4],
         validator_configs: make_identical_validator_configs(&validator_config, 4),
         slots_per_epoch,
-        stakers_slot_offset: slots_per_epoch,
+        stakers_slot_offset,
+        ticks_per_slot,
         ..ClusterConfig::default()
     };
     let cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
     let client = RpcClient::new_socket(cluster.entry_point_info.rpc().unwrap());
 
-    assert!(client
-        .wait_for_max_stake(CommitmentConfig::default(), 33.0f32)
-        .is_ok());
+    // Expectation is it should only take a dozen or so epochs to warm up the
+    // necessary stake. This shouldn't take more than a minute or two with the
+    // given cluster parameters. Waiting 5 minutes provides plenty of margin.
+    let timeout = Some(Duration::from_secs(300));
+    if let Err(err) = client.wait_for_max_stake(CommitmentConfig::default(), 33.0f32, timeout) {
```

Comment:
nit: If we wanted to be more proper, we could derive the timeout from values above instead of hard-coding 300s. Ie, specify this in `ClusterConfig`
```rust
poh_config: PohConfig::new_sleep(Duration::from_micros(6250)),
```
and then calculate timeout as:
```rust
let num_expected_epochs = 12;
// Expect num_expected_epochs to be sufficient, but double it to provide some margin
// especially with things like tests running in parallel
let duration = config.poh_config.target_tick_duration * ticks_per_slot * slots_per_epoch * num_expected_epochs * 2;
```

Reply:
We'll still be vulnerable to any stake activation % changes, but I agree this is better. At least it will be easy to tell why we landed on the chosen number.